examples/custom-listeners/src/test/java/org/apache/james/examples/custom/listeners/BigMessageListenerTest.java

```diff
@@ -0,0 +1,203 @@
+            MessageManager.AppendCommand.builder()
+                .build(smallMessage()),
+            mailboxSession);
+
+        MessageResult addedMessage = messageIdManager
+            .getMessages(ImmutableList.of(composedIdOfSmallMessage.getMessageId()), FetchGroupImpl.MINIMAL, mailboxSession)
+            .get(0);
+        MessageMetaData oneMBMetaData = new MessageMetaData(addedMessage.getUid(), addedMessage.getModSeq(),
+            addedMessage.getFlags(), ONE_MB, addedMessage.getInternalDate(), addedMessage.getMessageId());
+
+        Event eventWithAFakeMessageSize = EventFactory.added()
+            .eventId(RANDOM_EVENT_ID)
+            .mailboxSession(mailboxSession)
+            .mailboxId(inboxId)
+            .mailboxPath(INBOX_PATH)
+            .addMetaData(oneMBMetaData)
+            .build();
+
+        testee.event(eventWithAFakeMessageSize);
+
+        assertThat(getMessageFlags(composedIdOfSmallMessage.getMessageId()))
+            .allSatisfy(flags -> assertThat(flags.contains(BIG_MESSAGE)).isTrue());
+    }
+
+    @Test
+    void listeningEventShouldNotRemoveOtherFlagsWhenBigMessages() throws Exception {
+        Flags appendMessageFlag = new Flags();
+        appendMessageFlag.add(Flags.Flag.SEEN);
+        appendMessageFlag.add(Flags.Flag.DRAFT);
+
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .withFlags(appendMessageFlag)
+                .build(bigMessage()),
+            mailboxSession);
+
+        assertThat(getMessageFlags(composedId.getMessageId()))
+            .allSatisfy(flags -> {
+                assertThat(flags.contains(Flags.Flag.SEEN)).isTrue();
+                assertThat(flags.contains(Flags.Flag.DRAFT)).isTrue();
+                assertThat(flags.contains(BIG_MESSAGE)).isTrue();
+            });
+    }
+
+    @Test
+    void listeningEventShouldKeepBigMessageFlagWhenAlreadySet() throws Exception {
+        Flags appendMessageFlag = new Flags();
+        appendMessageFlag.add(BIG_MESSAGE);
+
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .withFlags(appendMessageFlag)
+                .build(bigMessage()),
+            mailboxSession);
+
+        assertThat(getMessageFlags(composedId.getMessageId()))
+            .allSatisfy(flags -> assertThat(flags.contains(BIG_MESSAGE)).isTrue());
+    }
+
+    private Stream<Flags> getMessageFlags(MessageId messageId) throws Exception {
+        return messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, mailboxSession)
+            .stream()
+            .map(MessageResult::getFlags);
+    }
+
+    private Message bigMessage() throws Exception {
+        return Message.Builder.of()
+            .setSubject("big message")
+            .setBody(Strings.repeat("big message has size greater than one MB", 1024 * 1024), StandardCharsets.UTF_8)
```

Comment:
1024 * 1024 => ONE_MB

Reply:
The String `big message has size greater than one MB` is repeated 1024 * 1024 times. So It's absolutely bigger than 1 MB. And it fits the purpose of building a message having size > 1MB

Reply:
ok, i read it too fast.